lib/db/UserDAO.js

```diff
@@ -22,44 +37,57 @@ class UserDAO {
       return new Map();
     }
     const sql = 'SELECT * FROM "users" WHERE "id" IN ($(uniqueIds:csv))';
-    const rows = await db.manyOrNone(sql, { uniqueIds });
-    return new Map(rows.map(user => [user.id, user]));
+    const users = await db.manyOrNone(sql, { uniqueIds });
+    const usersNormalized = await normalizeUsers(users);
+    return new Map(usersNormalized.map(user => [user.id, user]));
   }
 
   static async bySub(sub) {
     const sql = 'SELECT * FROM "users" WHERE "sub" = $(sub)';
-    return db.oneOrNone(sql, { sub });
+    const user = await db.oneOrNone(sql, { sub });
+    return normalizeUser(user);
   }
 
   static async byEmail(email) {
     const sql = 'SELECT * FROM "users" WHERE "email" = $(email)';
-    return db.oneOrNone(sql, { email });
+    const user = await db.oneOrNone(sql, { email });
+    return normalizeUser(user);
   }
 
   static async create(user) {
     const sql = `
-INSERT INTO "users" ("createdAt", "email", "sub", "uniqueName")
-VALUES ($(createdAt), $(email), $(sub), $(uniqueName))
-RETURNING "id"`;
-    const createdAt = DateTime.local();
-    const { id } = await db.one(sql, {
-      createdAt,
-      ...user,
-    });
-    return {
-      id,
-      createdAt,
+INSERT INTO "users" (
+  "createdAt",
+  "email",
+  "scope",
+  "sub",
+  "uniqueName"
+) VALUES (
+  $(createdAt),
+  $(email),
+  $(scope),
+  $(sub),
+  $(uniqueName)
+) RETURNING "id"`;
+    const persistedUser = {
```

Comment:
I like that we're landing on a reliable CRUD pattern that incorporates both the transient vs. persisted language and Joi validation - makes these DAOs a lot more readable.